README.md

```diff
@@ -0,0 +1,205 @@
+hardware.
+
+This is achieved through a combination of kernel security features such as
+namespaces, mandatory access control and control groups.
+
+## Unprivileged Containers
+
+Unprivileged containers are containers that are run without any privilege. This
+requires support for user namespaces in the kernel that the container is run
+on. LXC was the first runtime to support unprivileged containers after user
+namespaces were merged into the mainline kernel.
+
+In essence, user namespaces isolate given sets of UIDs and GIDs. This is
+achieved by establishing a mapping between a range of UIDs and GIDs on the host
+to a different (unprivileged) range of UIDs and GIDs in the container. The
+kernel will translate this mapping in such a way that inside the container all
+UIDs and GIDs appear as you would expect from the host whereas on the host
+these UIDs and GIDs are in fact unprivileged. For example, a process running as
+UID and GID 0 inside the container might appear as UID and GID 100000 on the
+host. The implementation and working details can be gathered from the
+corresponding user namespace man page.
+
+Since unprivileged containers are a security enhancement they naturally come
+with a few restrictions enforced by the kernel. In order to provide a fully
+functional unprivileged container LXC interacts with 3 pieces of setuid code:
+
+- lxc-user-nic (setuid helper to create a veth pair and bridge it on the host)
+- newuidmap (from the shadow package, sets up a uid map)
+- newgidmap (from the shadow package, sets up a gid map)
+
+Everything else is run as your own user or as a uid which your user owns.
+
+In general, LXC's goal is to make use of every security feature available in
+the kernel. This means LXC's configuration management will allow experienced
+users to intricately tune LXC to their needs.
+
+A more detailed introduction into LXC security can be found under the following link
+
+- https://linuxcontainers.org/lxc/security/
+
+### Removing all Privilege
+
+In principle LXC can be run without any of these tools provided the correct
+configuration is applied. However, the usefulness of such containers is usually
+quite restricted. Just to highlight the two most common problems:
+
+1. Network: Without relying on a setuid helper to setup appropriate network
+   devices for an unprivileged user (see LXC's `lxc-user-nic` binary) the only
+   option is to share the network namespace with the host. Although this should
+   be secure in principle, sharing the host's network namespace is still one
+   step of isolation less and increases the attack vector.
+   Furthermore, when host and container share the same network namespace the
+   kernel will refuse any sysfs mounts. This usually means that the init binary
+   inside of the container will not be able to boot up correctly.
+
+2. User Namespaces: As outlined above, user namespaces are a big security
+   enhancement. However, users which are unprivileged on the host will only be
```

Comment:
Oh, sorry - this sentence is confusing.  Can you turn that around, something like

"However, without relying on privileged helpers users who are unprivileged on the host are only permitted to map their own UID into a container."